index.bs

```diff
@@ -2153,6 +2154,68 @@ It is recommended that once [=explicit consent=] is granted for a specific [=/or
+
+### Limiting ### {#limiting}
+Limiting is when data is reported only when it is within a specific range. For example, it is possible to comfortably limit reporting positional pose data when a user has moved beyond a specific distance away from an approved location. Care should be taken to ensure that the user experience is not negatively affected when employing this mitigation. It is often desireable to avoid a 'hard stop' at the at the end of a range as this may cause disruptive user experiences.
+
+Protected functionality {#protected-functionality}
+-----------------------
+
+The [=sensitive information=] exposed by the API can be divided into categories that share threat profiles and necessary protections against those threats.
+
+### Immersiveness ### {#protect-immersiveness}
+Users must be in control of when immersive sessions are created because the creation causes invasive changes on a user's machine. For example, starting an [=immersive session=] will engage the [=/XR device=] sensors, take over access to the device's display, and begin presentating immersive content which may terminate another application's access to the XR hardware. It may also incur significant power or performance overhead on some systems or trigger the launching of a status tray or storefront.
+
+<div class="algorithm" data-algorithm="immersive-session-allowed">
+
+To determine if an <dfn>immersive session request is allowed</dfn> the user agent MUST run the following steps:
+
+  1. If the request was not [=triggered by user activation=], return <code>false</code>
+  1. If the request does not originate from a [=trustworthy=] document and origin, return <code>false</code>
+  1. If the request does not originate from a document that is [=active and focused=], return <code>false</code>
+  1. If the request does not originates from a document that is allowed to use the WebXR [[#feature-policy|feature policy]] as well as the [=underlying sensors=]' feature policies, return <code>false</code>
+  1. If [=user intent=] is not well understood, either via [=explicit consent=] or [=implicit consent=], return <code>false</code>
+  1. Return <code>true</code>
+
+</div>
+
+### Poses ### {#protect-poses}
+When based on sensor data, {{XRPose}} and {{XRViewerPose}} will expose [=sensitive information=] that may be misused in a number of ways, including input sniffing, gaze tracking, or fingerprinting.
+
+<div class="algorithm" data-algorithm="poses-allowed">
+
+To determine if <dfn>poses may be reported</dfn> to an {{XRSession}} |session|, the user agent MUST run the following steps:
```

Comment:
I disagree with the suggestion in this case specifically because we're defining an algorithm here that is explicitly called by the algorithm which populates the pose, which implies it'll be called every time.